hal/mbed_pinmap_default.c

```diff
@@ -0,0 +1,239 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "hal/pinmap.h"
+#include "platform/mbed_toolchain.h"
+#include "platform/mbed_assert.h"
+#include "PeripheralPins.h"
+
+//*** Common form factors ***
+#ifdef TARGET_FF_ARDUINO
```

Comment:
Should be somthing like MBED_CONF_TARGET_DEFAULT_FORM_FACTOR==ARDUINO ?

Reply:
I was thinking that all the form factors a board supports can be present at the same time. So for example ST boards can have both `TARGET_FF_MORPHO` and `TARGET_FF_ARDUINO`. Setting the value of `MBED_CONF_TARGET_DEFAULT_FORM_FACTOR` would just pick one of these to be used during testing. What do you think @jeromecoutant

Reply:
To be honest, for me, TARGET_FF_MORPHO and TARGET_FF_ARDUINO are not used at all...

So be free to make the changes you need :-)